gtsfm/frontend/retriever/default_retriever.py

```diff
@@ -0,0 +1,62 @@
+"""Default image retriever for front-end.
+Authors: Travis Driver, Jon Womack
+"""
+import abc
+from typing import List, Optional, Tuple, Dict
+import itertools
+
+import numpy as np
+import cbir
+import dask
+from dask.delayed import Delayed
+from gtsam import Pose3, Cal3Bundler
+import networkx as nx
+import pickle
+
+from gtsfm.frontend.retriever.retriever_base import RetrieverBase
+import gtsfm.utils.logger as logger_utils
+from gtsfm.common.keypoints import Keypoints
+
+
+logger = logger_utils.get_logger()
+
+
+class DefaultRetriever(RetrieverBase):
+    """Default class for image retrieval. Keeps all image pairs from Dataloader.
+    """
+
+    def __init__(self, image_pair_indices):
+        """Initialize the Retriever.
+        Args:
+        """
+        super().__init__(image_pair_indices)
+
+    def retrieve_potential_matches(self, i1, i2, image_graph) -> Delayed:
```

Comment:
My suggestion -- provide to this function the loader (or pass all images and all image indices).

Then return all suggested image pairs

Reply:
I've passed all images. I currently have the image indices being passed upon retriever initialization in base retriever (all retrievers will select a subset of possible image pairs from Dataloader).